sdk/monitor/monitor-query/samples-dev/metricsQuery.ts

```diff
@@ -24,26 +24,33 @@ export async function main() {
   const iterator = metricsQueryClient.listMetricDefinitions(metricsResourceId);
   let result = await iterator.next();
   const firstMetric: MetricDefinition = result.value;
-
+  let secondMetricName: string;
   while (!result.done) {
     console.log(` metricDefinitions - ${result.value.id}, ${result.value.name}`);
+    secondMetricName = result.value.name!;
     result = await iterator.next();
   }
   console.log(`First Metric Definition = ${firstMetric.name}`);
 
   console.log(`Picking an example metric to query: ${firstMetric.name!}`);
 
-  const metricsResponse = await metricsQueryClient.query(metricsResourceId, [firstMetric.name!], {
-    granularity: "PT1M",
-    timespan: { duration: Durations.FiveMinutes }
-  });
+  const metricsResponse = await metricsQueryClient.query(
+    metricsResourceId,
+    [firstMetric.name!, secondMetricName!],
```

Comment:
Can `name` really be undefined? Is that just because of the swagger model, or can the service really return an undefined name? Using the non-null assertion always makes me suspicious.

Reply:
bcz of swagger model -  Metric Definitions and Metric Namespaces can return empty objects as specified in the swagger by the service.

Reply:
I get that the swagger model models the name as optional, but my question is, does the service actually treat name as optional or is it a mistake in the model? We're using the non-null assertion in our sample, which either means we believe it _is_ actually always defined - so why make it optional - or we're not showing the customer how to handle a legitimate edge case.

If `name` was actually undefined, would that be an error? What happens if you end up passing `undefined` to `query`?

Reply:
if we pass undefined to query the service will return an error - so we shouldn't do that. I updated the sample to do the verification of both metric names being valid